### DISCUSSION TITLE
API for archiving a notification (marking as done)

### DISCUSSION THREAD

**Discussion by aspiers** [community/community#50224] [Apps, API and Webhooks]
### Select Topic Area

Product Feedback

### Body

The [GitHub website](https://github.com/notifications) and mobile app are both able to mark individual notifications as done; however there does not appear to be any API mechanism documented which allows this, either in the REST API or the GraphQL API.

The REST API only allows marking notifications as *read*, not as *done*:

https://docs.github.com/en/rest/activity/notifications?apiVersion=2022-11-28#mark-notifications-as-read

and I can't find anything about notifications in the GraphQL API *at all* (maybe they're there, but just not easily discoverable).

However, sniffing the network traffic when archiving a notification from https://github.com/notifications reveals a beta API:

![image](https://user-images.githubusercontent.com/100738/225621499-28345299-8c19-4a8c-a8db-8c9382813fb9.png)

![image](https://user-images.githubusercontent.com/100738/225621628-1b81c0ab-9175-4437-9322-6a5c46a9bd23.png)

It would be great if this was brought out of beta and fully documented!

**Comment by morgen-peschke** [comment]
This would be particularly nice as the current UI does not have great UX for tasks like "archive all notifications for closed PRs"

**Reply by andrew** [reply]
You can make a saved search for `type:pull_request state:merged,closed` which will give you a fairly quick way of archiving them all. I used `state:merged,closed inbox:true starred:false` as a search I called "Archivable"

**Comment by joshbeard** [comment]
Maybe this is implied, but it'd also be helpful to filter on this status when retrieving notifications. The UI currently defaults to "not done" notifications, distinctly separate from "unread".

**Comment by andrew** [comment]
@aspiers I took a look at their beta endpoint but I believe it requires a different kind of authentication token than is provided by the official api so we can't use the unofficial one as a hack, will need to wait for it to be implemented.

@joshbeard yes if we can retrieve the "done" state from the api we can provide filters and searches on that field. 

If you'd like to see this feature I recommend contacting github customer support and requesting the api be made public, the more people that ask, the most likely it will happen.

**Reply by pradyunsg** [reply]
https://support.github.com/contact?tags=rr-general-technical is the current relevant URL, to reach out to customer support for "general technical" stuff by the looks of it.

**Comment by rmacklin** [comment]
This API has been introduced: https://docs.github.com/en/rest/activity/notifications?apiVersion=2022-11-28#mark-a-thread-as-done

However, there still doesn't seem to be an API parameter to filter notifications using `is:done` (or not) like we can in the web UI. This makes it difficult to actually make use of the "mark as done" API for integrations like gh-notify: https://github.com/meiji163/gh-notify/issues/47#issuecomment-1974962747

**Reply by thehesiod** [reply]
I was hoping to write a script to mark all my not-done notifications done, however the notifications list api returns done notifications as well :(

**Reply by mkurz** [reply]
It would be great if we could filter for "done" notifications, and also all notications results should contain a timestampe field `done`, which when not null, should contain the exact timestamp when a nofications was marked as done

**Comment by balusena** [comment]
Create a saved search with "type:pull_request state:merged,closed" for quick archiving. Alternatively, use "state:merged,closed inbox:true starred:false" as a search termed "Archivable" for efficient management of pull requests.

**Reply by morgen-peschke** [reply]
This appears to be broken at the moment. 

https://github.com/notifications?query=type%3Apull_request+state%3Amerged%2Cclosed and https://github.com/notifications?query=type%3Apull_request+state%3Amerged return results which include open PRs

**Comment by james-elicx** [comment]
I'm surprised the notification APIs haven't been productionised yet. Feels bizarre to have the API endpoint in beta for 3 years, that's some serious tech debt... Would love to have this...

**Comment by drbenvincent** [comment]
Came here to request this functionality too

**Comment by Alwaysin** [comment]
This is now possible as of API version 2026-03-10: https://docs.github.com/en/rest/activity/notifications?apiVersion=2026-03-10#mark-a-thread-as-done

**Reply by rmacklin** [reply]
It has been possible for over two years, but there is a related gap in filtering out Done items: https://github.com/orgs/community/discussions/50224#discussioncomment-8658939
